platform-core-module/src/main/java/com/softicar/platform/core/module/file/stored/content/StoredFileContentUploader.java

```diff
@@ -32,35 +33,43 @@ class StoredFileContentUploader {
 	private static final String TEMPORARY_FILE_FOLDER = "/tmp";
 
 	private final IStoredFileDatabase database;
-	private final IStoredFileContentStore store;
 	private final AGStoredFile storedFile;
+	private final Optional<IStoredFileContentStore> store;
 
-	public StoredFileContentUploader(IStoredFileDatabase database, IStoredFileContentStore store, AGStoredFile storedFile) {
+	public StoredFileContentUploader(IStoredFileDatabase database, AGStoredFile storedFile, IStoredFileContentStore store) {
+
+		this(database, storedFile, Optional.ofNullable(store));
+	}
+
+	public StoredFileContentUploader(IStoredFileDatabase database, AGStoredFile storedFile, Optional<IStoredFileContentStore> store) {
 
 		this.database = database;
-		this.store = store;
 		this.storedFile = storedFile;
+		this.store = store;
 	}
 
 	public OutputStream createOutputStream() {
 
-		if (!store.isEnabled()) {
+		if (!store.isPresent()) {
 			return useChunks();
 		}
 
-		if (store.isReady()) {
-			long freeDiskSpace = store.getFreeDiskSpace();
+		if (store.get().isAvailable()) {
```

Comment:
Abundant use of `store.get()` here and below should probably be fixed / beautified.